File: OOP based implmentation/OOP based implmentation/Project/objs/StudentPersonal.cpp
#include <string>
#include <iostream>
#include <fstream>
#include "StudentPersonal.h"

using namespace std;

StudentPersonal::StudentPersonal(const string &name, int age, const string &gender, const string &address, const string &hobby)
{
    this->name = name;
    this->age = age;
    this->gender = gender;
    this->address = address;
    this->hobby = hobby;
}

string StudentPersonal::getName() const
{
    return name;
}

int StudentPersonal::getAge() const
{
    return age;
}

string StudentPersonal::getGender() const
{
    return gender;
}

string StudentPersonal::getAddress() const
{
    return address;
}

string StudentPersonal::getHobby() const
{
    return hobby;
}

void StudentPersonal::writeToCSV(ofstream &csv)
{
    csv << name << "," << age << "," << gender << "," << address << "," << hobby << "\n";
}

File: OOP based implmentation/OOP based implmentation/Project/objs/StudentPersonal.java
package objs;
import java.io.BufferedWriter;
import java.io.IOException;

public class StudentPersonal {
    private String name;
    private int age;
    private String gender;
    private String address;
    private String hobby;
   public StudentPersonal(String name, int age, String gender, String address, String hobby) 
   {
   
        this.name = name;
        this.age = age;
        this.gender = gender;
        this.address = address;
        this.hobby = hobby;
    }
    public String getName()
    {
        return name;
    }
    public int getAge()
    {
        return age;
    }
    public String getGender()
    {
        return gender;
    }
    public String getAddress()
    {
        return address;
    }
    public String getHobby()
    {
        return hobby;
    }
    public void writeToCSV(BufferedWriter csv) throws IOException {
        csv.write(name + "," + age + "," + gender + "," + address + "," + hobby + "\n");
    }
}
